### ISSUE TITLE
Trigger rerun on pkg-config probed mysql library version bump

### ISSUE THREAD

**Issue by Banyc** [sgrif/mysqlclient-sys#58]
Currently, [pkt-config probed library path](https://github.com/sgrif/mysqlclient-sys/blob/a1799fca97ce0016d12c36095f0884426bdffed8/build.rs#L293) are not monitored by [cargo change detection](https://doc.rust-lang.org/cargo/reference/build-scripts.html#rerun-if-changed). Once a user upgraded their mysql library via homebrew without `cargo clean`, this is the resulting linking error:

```text
  = note: ld: warning: search path '/opt/homebrew/Cellar/mysql/9.0.1_4/lib' not found
          ld: library 'mysqlclient' not found
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

At this moment, the library is at a new location:

```text
$ pkg-config --libs mysqlclient
-L/opt/homebrew/Cellar/mysql/9.0.1_6/lib -lmysqlclient
```

Solution may be to `cargo::rerun-if-changed` on the directory where the library resides.

**Comment by weiznich** [comment]
Thanks for reporting. I'm happy to merge a PR that adds this `cargo::rerun-if-changed` statement to the buildscript.

**Comment by Banyc** [comment]
I made this snippet:

```rust
const PKG_CONFIG_MYSQL_LIB: &str = "mysqlclient";

#[cfg(test)]
#[test]
fn test_pkg_out() {
    if let Ok(lib) = pkg_config::probe_library(PKG_CONFIG_MYSQL_LIB) {
        dbg!(&lib);
    }
}
```

...and it returns:

```text
cargo:rerun-if-env-changed=MYSQLCLIENT_NO_PKG_CONFIG
cargo:rerun-if-env-changed=PKG_CONFIG
cargo:rerun-if-env-changed=MYSQLCLIENT_STATIC
cargo:rerun-if-env-changed=MYSQLCLIENT_DYNAMIC
cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
cargo:rerun-if-env-changed=PKG_CONFIG_PATH
cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
cargo:rerun-if-env-changed=MYSQLCLIENT_STATIC
cargo:rerun-if-env-changed=MYSQLCLIENT_DYNAMIC
cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
cargo:rustc-link-search=native=/opt/homebrew/Cellar/mysql/9.0.1_6/lib
cargo:rustc-link-lib=mysqlclient
cargo:rerun-if-env-changed=PKG_CONFIG
cargo:rerun-if-env-changed=MYSQLCLIENT_STATIC
cargo:rerun-if-env-changed=MYSQLCLIENT_DYNAMIC
cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
cargo:rerun-if-env-changed=PKG_CONFIG_PATH
cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
[src/lib.rs:7:9] &lib = Library {
    libs: [
        "mysqlclient",
    ],
    link_paths: [
        "/opt/homebrew/Cellar/mysql/9.0.1_6/lib",
    ],
    link_files: [],
    frameworks: [],
    framework_paths: [],
    include_paths: [
        "/opt/homebrew/Cellar/mysql/9.0.1_6/include/mysql",
    ],
    ld_args: [],
    defines: {},
    version: "24",
    _priv: (),
}
```

Either watching `link_paths` or `include_paths` seems reasonable and I can't decide which one to actually `cargo::rerun` on. What is your suggestion? @weiznich

**Comment by Banyc** [comment]
I ended up chose `link_paths` because it sits the actual library binary and pushed a PR.

**Comment by weiznich** [comment]
Closed by #59

**Comment by weiznich** [comment]
Published as https://github.com/sgrif/mysqlclient-sys/releases/tag/v0.4.2
